### PR TITLE
Refactor Role Skills Controller Tests

### LINKED ISSUE #414: Refactor Role Skill Controller tests
# Problem

Adding api_case.ex helpers to refactors controller tests for Role skills

## References

Progress on: #413

### PR DESCRIPTION
**Pull request by amyschools**
# What's in this PR?

Adding helpers from api_case.ex.

## References
Fixes #414 

Progress on: #413

### CODE REVIEW

**Review by sbatson5 — APPROVED** [review]
This looks good to me 👍. Nice work!

File: test/controllers/role_skill_controller_test.exs [hunk 1]
```diff
@@ -1,83 +1,42 @@
...
-  defp build_payload, do: %{ "data" => %{"type" => "role-skill", "attributes" => %{}}}
+  use CodeCorps.ApiCase, resource_name: :role_skill
 
   describe "index" do
     test "lists all entries on index", %{conn: conn} do
       path = conn |> role_skill_path(:index)
-      json = conn |> get(path) |> json_response(200)
-      assert json["data"] == []
+      conn = conn |> get(path)
+      assert json_response(conn, 200)["data"] == []
```
Comment (joshsmith) [inline test/controllers/role_skill_controller_test.exs]: I would keep this roughly as it is in some of the closed PRs. `conn |> get(path) |> json_response(200)`, then assign the response and assert against that.

File: test/controllers/role_skill_controller_test.exs [hunk 2]
```diff
@@ -87,62 +46,44 @@ defmodule CodeCorps.RoleSkillControllerTest do
       role = insert(:role, name: "Frontend Developer")
       skill = insert(:skill, title: "test skill")
```
Comment (joshsmith) [inline test/controllers/role_skill_controller_test.exs]: I don't think you need to the attributes here. Just `insert(:role)` will work.

**Review by joshsmith — CHANGES_REQUESTED** [review]

File: test/controllers/role_skill_controller_test.exs (hunk 1 above)
Comment (amyschools) [reply test/controllers/role_skill_controller_test.exs]: is this okay instead? i didn't realize there was a request_index function to help with that until just now.

```
describe "index" do
    test "lists all entries on index", %{conn: conn} do
    [project_skill_1, project_skill_2] = insert_pair(:project_skill)

     conn
     |> request_index
     |> json_response(200)
     |> assert_ids_from_response([project_skill_1.id, project_skill_2.id])
    end
```

Comment (joshsmith) [reply test/controllers/role_skill_controller_test.exs]: I didn't realize that either.

File: test/controllers/role_skill_controller_test.exs [hunk 3]
```diff
@@ -1,148 +1,88 @@
...
     test "lists all entries on index", %{conn: conn} do
-      path = conn |> role_skill_path(:index)
-      json = conn |> get(path) |> json_response(200)
-      assert json["data"] == []
+      [role_skill_1, role_skill_2] = insert_pair(:role_skill)
+
+       conn
+       |> request_index
+       |> json_response(200)
+       |> assert_ids_from_response([role_skill_1.id, role_skill_2.id])
```
Comment (joshsmith) [inline test/controllers/role_skill_controller_test.exs]: Spacing here is weird. There are three spaces.

File: test/controllers/role_skill_controller_test.exs [hunk 4]
```diff
@@ -1,148 +1,88 @@
...
-      second_result
-      |> assert_result_id(role_skill_2.id)
-      |> assert_jsonapi_relationship("role", role.id)
-      |> assert_jsonapi_relationship("skill", phoenix.id)
-
-      assert second_result["attributes"] == %{}
+      conn
+      |> get(path)
+      |> json_response(200)
+      |> assert_ids_from_response([role_skill_1.id, role_skill_2.id])
```
Comment (joshsmith) [inline test/controllers/role_skill_controller_test.exs:22]: I might use that `request_index` helper instead.

**Review by joshsmith — CHANGES_REQUESTED** [review]

File: test/controllers/role_skill_controller_test.exs (hunk 4 above)
Comment (amyschools) [reply test/controllers/role_skill_controller_test.exs:22]: That created identical tests for listing all the entries on the index and filtering the index...hm

Comment (amyschools) [reply test/controllers/role_skill_controller_test.exs:22]: The filter test should have a third item that's not included in the ids from response...maybe I should put that back in to test the filtering

Comment (joshsmith) [reply test/controllers/role_skill_controller_test.exs:22]: @amyschools yes, absolutely.

File: test/controllers/role_skill_controller_test.exs [hunk 5]
```diff
@@ -1,148 +1,88 @@
...
     test "filters resources on index", %{conn: conn} do
-      elixir = insert(:skill, title: "Elixir")
-      phoenix = insert(:skill, title: "Phoenix")
-      rails = insert(:skill, title: "Rails")
-
-      role = insert(:role)
-      role_skill_1 = insert(:role_skill, role: role, skill: elixir)
-      role_skill_2 = insert(:role_skill, role: role, skill: phoenix)
-      insert(:role_skill, role: role, skill: rails)
+      [role_skill_1, role_skill_2] = insert_pair(:role_skill)
```
Comment (begedin) [inline test/controllers/role_skill_controller_test.exs]: You figured out the missing third record, so I'll just link https://github.com/code-corps/code-corps-api/pull/425/files#r86388037 as a pointer on how to write this in one line.

**Review by begedin — APPROVED** [review]

**Review by joshsmith — APPROVED** [review]

**Pull request merged**